integration_test/gce/zone_picker.go

```diff
@@ -0,0 +1,61 @@
+package gce
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/smallnest/weighted"
+)
+
+// weightedRoundRobin implements a thread safe weighted round robin algorithm.
+// According to the `weighted` library, it is the same algorithm as the one
+// used by nginx to do weighted round robin.
+type weightedRoundRobin struct {
+	mutex sync.Mutex
+	// This type implements the weighted round robin algorithm. I don't know what
+	// SW stands for, maybe "smooth weighted".
+	sw *weighted.SW
+}
+
+// Next() selects the next zone to spawn a VM in.
+// It is thread safe.
+func (wrr *weightedRoundRobin) Next() string {
+	wrr.mutex.Lock()
+	defer wrr.mutex.Unlock()
+
+	return wrr.sw.Next().(string)
+}
+
+// parseZoneEnvironmentVariables looks at `weightedZones` (and `zone` if
+// needed) and extracts the zones and weights into a a weighted round robin
+// zone picker.
+// TODO(martijnvans): Remove support for `zone`.
+func newZonePicker(weightedZones, zone string) (*weightedRoundRobin, error) {
+	if weightedZones == "" {
+		if zone == "" {
+			return nil, errors.New("either ZONE or ZONES must be specified")
```

Comment:
Since we prefer `ZONES`, let's put it first?

```suggestion
			return nil, errors.New("either ZONES or ZONE must be specified")
```

Reply:
done, but I plan to get rid of ZONE anyway.